libs/community/langchain_community/llms/bedrock.py

```diff
@@ -318,6 +362,55 @@ def _prepare_input_and_invoke_stream(
             if run_manager is not None:
                 run_manager.on_llm_new_token(chunk.text, chunk=chunk)
 
+    async def _aprepare_input_and_invoke_stream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[AsyncCallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> AsyncIterator[GenerationChunk]:
+        _model_kwargs = self.model_kwargs or {}
+        provider = self._get_provider()
+
+        if stop:
+            if provider not in self.provider_stop_sequence_key_name_map:
+                raise ValueError(
+                    f"Stop sequence key name for {provider} is not supported."
+                )
+            _model_kwargs[self.provider_stop_sequence_key_name_map.get(provider)] = stop
+
+        if provider == "cohere":
+            _model_kwargs["stream"] = True
+
+        params = {**_model_kwargs, **kwargs}
+        input_body = LLMInputOutputAdapter.prepare_input(provider, prompt, params)
+        body = json.dumps(input_body)
+
+        loop = asyncio.get_running_loop()
+
+        # Use ThreadPoolExecutor to run the synchronous method in a separate thread
+        with ThreadPoolExecutor() as executor:
```

Comment:
Is there a need for using a thread pool here, given the task could be run directly with asyncio?
```python
await asyncio.get_running_loop().run_in_executor(
    None, 
    lambda: self.client.invoke_model_with_response_stream(
            body=body,
            modelId=self.model_id,
            accept="application/json",
            contentType="application/json",
        )
)
```

Reply:
Good point, I'll modify that and rebase for a review today/tomorrow. Thanks @3coins !